torch_geometric/distributed/dist_neighbor_loader.py

```diff
@@ -0,0 +1,154 @@
+import logging
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+import torch
+
+from torch_geometric.data import Data, FeatureStore, GraphStore, HeteroData
+from torch_geometric.loader.node_loader import NodeLoader
+from torch_geometric.loader.utils import filter_custom_store
+from torch_geometric.sampler import HeteroSamplerOutput, SamplerOutput
+from torch_geometric.sampler.base import SubgraphType
+from torch_geometric.typing import EdgeType, InputNodes, OptTensor, as_str
+
+from .dist_context import DistContext, DistRole
+from .dist_loader import DistLoader
+from .dist_neighbor_sampler import DistNeighborSampler
+from .local_feature_store import LocalFeatureStore
+from .local_graph_store import LocalGraphStore
+
+
+class DistNeighborLoader(NodeLoader, DistLoader):
+    r""" A distributed loader that preform sampling from nodes.
+    Args:
+
+      """
+    def __init__(
+            self,
+            data: Tuple[LocalFeatureStore, LocalGraphStore],
+            num_neighbors: Union[List[int], Dict[EdgeType, List[int]]],
+            master_addr: str,
+            master_port: Union[int, str],
+            current_ctx: DistContext,
+            rpc_worker_names: Dict[DistRole, List[str]],
+            neighbor_sampler: Optional[DistNeighborSampler] = None,
+            input_nodes: InputNodes = None,
+            input_time: OptTensor = None,
+            replace: bool = False,
+            subgraph_type: Union[SubgraphType, str] = 'directional',
+            disjoint: bool = False,
+            temporal_strategy: str = 'uniform',
+            time_attr: Optional[str] = None,
+            transform: Optional[Callable] = None,
+            transform_sampler_output: Optional[Callable] = None,
+            is_sorted: bool = False,
+            directed: bool = True,  # Deprecated.
+            with_edge: bool = False,
+            concurrency: int = 1,
+            collect_features: bool = True,
+            filter_per_worker: Optional[bool] = False,
+            async_sampling: bool = True,
+            device: torch.device = torch.device('cpu'),
+            **kwargs,
```

Comment:
Do we need to support all of these arguments right away? Would prefer to shrink them down at first (e.g., I am not sure if we want to integrate something like temporal sampling right away?).

Reply:
As far as I know, temporal sampling is already supported in DistSampler. The arguments follow the patter of standard PyG loaders. Please let me know if that's ok or should we shorten this list.